index/scorch/scorch.go

```diff
@@ -376,6 +375,8 @@ func (s *Scorch) Delete(id string) error {
 func (s *Scorch) Batch(batch *index.Batch) (err error) {
 	start := time.Now()
 
+	// notify handlers that we're about to introduce a segment
```

Comment:
nit: notify handlers that we're about to index a batch of data, sounds better perhaps

(apart from this looks good to me)